indexer/src/main.rs

```diff
@@ -84,11 +97,7 @@ fn read_config<T: serde::de::DeserializeOwned>(
         let config_str = std::fs::read_to_string(config_path)?;
         serde_yaml::from_str(&config_str).map_err(Into::into)
     } else {
-        if let Some(config_args) = config_args {
-            Ok(config_args)
-        } else {
-            panic!("Either config_path or config_args must be provided")
-        }
+        config_args.ok_or_else(|| Error::AnyhowError(anyhow::anyhow!("Either config_path or config_args must be provided")))
```

Comment:
p sure just doing Err(anyhow!( works well here, but you also have eyre imported which works with more errors that are into dyn Err